### PR TITLE
Web v2-demo

### PR DESCRIPTION
**Pull request by kamikos**
working frontend demo as of 9th May

### CODE REVIEW

**Comment by papaj-na-wrotkach** [comment]
@kamikos check my changes, when s1 approves it we'll merge it

File: server/lib/PasswordUtils.hh [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+
+class PasswordUtils
+{
+  public:
+    static std::string
+    hashPassword(std::string const& password);
+
+    static bool
+    checkHash(std::string const& hash, std::string const& password);
+};
```
Comment (Xertes0) [inline server/lib/PasswordUtils.hh]: Is there any need for a class here? Those functions would be better in a namespace.

File: server/lib/PasswordUtils.cc [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+      1,
+      65536, // TODO(papaj-na-wrotkach): move to config
+      1);
+}
+
+bool
+PasswordUtils::checkHash(std::string const& hash, std::string const& password)
+{
+    return Botan::argon2_check_pwhash(password.c_str(), password.size(), hash);
+}
```
Comment (Xertes0) [inline server/lib/PasswordUtils.cc]: Botan takes `char const*`, not an `std::string`, so it'd be better to take `std::string_view`s as arguments. By doing this we avoid a situation where we only have a `char const*`, and have to unnecessarily convert it to `std::string`.

And yes, `std::string_view` has a size function.

**Review by Xertes0 — CHANGES_REQUESTED** [review]
Other than what I commented on, it's OK.

File: server/lib/PasswordUtils.cc (hunk 2 above)
Comment (Xertes0) [reply server/lib/PasswordUtils.cc]: To elaborate,
`std::string_view` does not make any dynamic memory allocations, but only holds a pointer to the begging, and size of the string. That's why it's preferred for read-only string 'views'.

**Review by Xertes0 — APPROVED** [review]
LGTM :+1:

**Pull request merged**